src/main/java/core/basesyntax/db/HandlerStorage.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.FruitRecord;
+import core.basesyntax.strategy.BalanceOperationHandlerImpl;
+import core.basesyntax.strategy.DecreaseAmountOperationHandlerImpl;
+import core.basesyntax.strategy.IncreaseAmountOperationHandlerImpl;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public class HandlerStorage {
+    private static final Map<FruitRecord.Operation, OperationHandler> handlerMap;
+
+    static {
+        handlerMap = new HashMap<>();
+        handlerMap.put(FruitRecord.Operation.PURCHASE, new DecreaseAmountOperationHandlerImpl());
+        handlerMap.put(FruitRecord.Operation.SUPPLY, new IncreaseAmountOperationHandlerImpl());
+        handlerMap.put(FruitRecord.Operation.BALANCE, new BalanceOperationHandlerImpl());
+        handlerMap.put(FruitRecord.Operation.RETURN, new IncreaseAmountOperationHandlerImpl());
+    }
```

Comment:
let's create handlerMap in main method and pass it to HandlerStorage class in constructor. Than we can pass HandlerStorage instance in FruitShopService class (also through constructor)

Reply:
fixed